cvrf2csaf/section_handlers/document_acknowlegments.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from ..common.common import SectionHandler
+
+
+class Acknowledgments(SectionHandler):
+    def __init__(self):
+        super().__init__()
+
+    def _process_mandatory_elements(self, root_element):
+
+        if hasattr(root_element, 'Acknowledgment'):
+            # at least one entry in list of Acknowledgments
+            pass # No field is mandatory due to CVRF.xsd; version 1.2
+
+
+    def _process_optional_elements(self, root_element):
+        # optional values
+        names = []
+        organization = []
+        summary = []
+        urls = []
+        for ack in root_element.Acknowledgment:
+            names.append(ack.Name.text)
+            if hasattr(ack, 'Organization'):
+                organization.append(ack.Organization.text)
+            if hasattr(ack, 'Summary'):
+                summary.append(ack.Summary.text)
+            if hasattr(ack, 'Url'):
+                urls.append(ack.Url.text)
+
+        self.csaf['names'] = names
```

Comment:
See https://github.com/csaf-tools/CVRF-CSAF-Converter/pull/33#discussion_r784794464 and https://github.com/csaf-tools/CVRF-CSAF-Converter/pull/33#discussion_r784927598: Empty arrays shall not be submitted to the final CSAF.

Reply:
Removing empty arrays can easily be done in the final dict:

`csaf = {k:v for k,v in csaf.items() if v}`

This would be prevent us from checking the length on each and every function.

But sure, integrating a length check here as well.